data-prepper-plugins/saas-source-plugins/source-crawler/src/main/java/org/opensearch/dataprepper/plugins/source/source_crawler/coordination/state/CrowdStrikeLeaderProgressState.java

```diff
@@ -0,0 +1,22 @@
+package org.opensearch.dataprepper.plugins.source.source_crawler.coordination.state;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+import org.opensearch.dataprepper.plugins.source.source_crawler.base.LeaderProgressState;
+
+import java.time.Instant;
+
+@Data
+public class CrowdStrikeLeaderProgressState implements LeaderProgressState {
+
+    @JsonProperty("last_poll_time")
+    private Instant lastPollTime;
+
+    @JsonProperty("remaining_days")
+    private int remainingDays;
+
+    public CrowdStrikeLeaderProgressState(@JsonProperty("last_poll_time") final Instant lastPollTime, @JsonProperty("remaining_days") int remainingDays) {
```

Comment:
Nit: do we need the `@JsonProperty` annotation on both the constructor parameters and the member variables? Jackson should be able to deserialize with just a no-args constructor + setters on the member variables